config/configmapprovider/file.go

```diff
@@ -22,18 +22,18 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-type fileMapProvider struct {
+type FileMapProvider struct {
 	fileName string
 }
 
-// NewFile returns a new Provider that reads the configuration from the given file.
-func NewFile(fileName string) Provider {
-	return &fileMapProvider{
+// NewFile returns a new MapProvider that reads the configuration from the given file.
+func NewFile(fileName string) *FileMapProvider {
+	return &FileMapProvider{
 		fileName: fileName,
 	}
 }
 
-func (fmp *fileMapProvider) Retrieve(_ context.Context, _ func(*ChangeEvent)) (Retrieved, error) {
+func (fmp *FileMapProvider) Retrieve(_ context.Context, _ func(*ChangeEvent)) (RetrievedMap, error) {
```

Comment:
Why do you need this if the other call can retrieve a map? That was my whole concern that we don't need 2 concepts we just need to find the "middle ground" between the full flexibility ConfigSource have and restricted scope of MapProvider. The reality is that the "ConfigSources" for example do not use params except for the "template" case for example, so we need to answer is that a reasonable use-case so we need to support params?